wpiunits/src/main/java/edu/wpi/first/units/Measure.java

```diff
@@ -299,11 +299,9 @@ default boolean isNear(Measure<U> other, Measure<U> tolerance) {
    * @return true if this measure is equivalent, false otherwise
    */
   default boolean isEquivalent(Measure<?> other) {
-    if (!this.unit().getBaseUnit().equals(other.unit().getBaseUnit())) {
-      return false; // Disjoint units, not compatible
-    }
-
-    return Math.abs(baseUnitMagnitude() - other.baseUnitMagnitude()) <= EQUIVALENCE_THRESHOLD;
+    // Check for disjoint units that aren't compatible
+    return this.unit().getBaseUnit().equals(other.unit().getBaseUnit())
+        && Math.abs(baseUnitMagnitude() - other.baseUnitMagnitude()) <= EQUIVALENCE_THRESHOLD;
```

Comment:
A bit late, but since the comment was previously applied to the false case, I'm not sure the new comment is accurate.

Reply:
Yeah, the comment is confusing now. It should say `Return false for disjoint units that aren't compatible`.

Reply:
Fixed in #6727, good to be resolved. (I don't have permissions to resolve this since I don't have write access and I'm not the PR author)